config/peerpods/podvm/libvirt-podvm-image-handler.sh

```diff
@@ -0,0 +1,280 @@
+
+    # Upload the created qcow2 to the volume
+    upload_libvirt_image
+
+    # Add the libvirt_volume_name to peer-pods-cm configmap
+    add_libvirt_vol_to_peer_pods_cm
+}
+
+# Function to dowload the rhel base image
+
+function download_rhel_kvm_guest_qcow2() {
+    ARCH=$(uname -m)
+    export ARCH
+
+    # Define the API endpoints
+    TOKEN_GENERATOR_URI=https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token
+    IMAGES_URI=https://api.access.redhat.com/management/v1/images/rhel/"${BASE_OS_VERSION}"/"${ARCH}"
+
+    filename="rhel-"${BASE_OS_VERSION}"-"${ARCH}"-kvm.qcow2"
+
+    token=$(curl "${TOKEN_GENERATOR_URI}" \
+        -d grant_type=refresh_token -d client_id=rhsm-api -d refresh_token="${REDHAT_OFFLINE_TOKEN}" | jq --raw-output .access_token)
+    images=$(curl -X 'GET' "${IMAGES_URI}" \
+        -H 'accept: application/json' -H "Authorization: Bearer "${token}"" | jq )
+
+    download_href=$(echo "${images}" | jq -r --arg fn "${filename}" '.body[] | select(.filename == $fn) | .downloadHref')
+
+    download_url=$(curl -X 'GET' "${download_href}" \
+        -H "Authorization: Bearer "${token}"" -H 'accept: application/json' | jq -r .body.href )
+
+    curl -X GET "${download_url}" -H "Authorization: Bearer "${token}"" --output rhel-"${BASE_OS_VERSION}"-"${ARCH}"-kvm.qcow2
+
+    cp -pr rhel-"${BASE_OS_VERSION}"-"${ARCH}"-kvm.qcow2 "${CAA_SRC_DIR}"/podvm/rhel-"${BASE_OS_VERSION}"-"${ARCH}"-kvm.qcow2
+
+    export IMAGE_URL="${CAA_SRC_DIR}"/podvm/rhel-"${BASE_OS_VERSION}"-"${ARCH}"-kvm.qcow2
+    export IMAGE_CHECKSUM=$(sha256sum "${IMAGE_URL}" | awk '{ print $1 }')
+
+}
+
+# Function to upload the qcow2 image to volume
+
+function upload_libvirt_image() {
+    echo "LIBVIRT_VOL_NAME: "${LIBVIRT_VOL_NAME}"" && echo "LIBVIRT_POOL: "${LIBVIRT_POOL}"" && \
+	    echo "LIBVIRT_URI: "${LIBVIRT_URI}"" && echo "PODVM_IMAGE_PATH: "${PODVM_IMAGE_PATH}"" 
+    echo "Starting to upload the image."
+    virsh -d 0 -c "${LIBVIRT_URI}" vol-upload --vol "${LIBVIRT_VOL_NAME}" "${PODVM_IMAGE_PATH}" --pool "${LIBVIRT_POOL}" --sparse
+    if [ $? -eq 0 ]; then
+        echo "Uploaded the image successfully"
+    fi
+}
+
+# Function to add the libvirt_volume_name in the peer-pods-cm configmap
+
+function add_libvirt_vol_to_peer_pods_cm(){
+    echo "Adding libvirt volume to peer-pods-cm configmap"
+
+    # Check if the peer-pods-cm configmap exists
+    if ! kubectl get configmap peer-pods-cm -n openshift-sandboxed-containers-operator >/dev/null 2>&1; then
+        echo "peer-pods-cm configmap does not exist. Skipping adding the libvirt volume name"
+        return
+    fi
+
+    # Add the libvirt image id to peer-pods-cm configmap
+    kubectl patch configmap peer-pods-cm -n openshift-sandboxed-containers-operator \
+    --type merge -p "{\"data\":{\"LIBVIRT_IMAGE_ID\":\"${LIBVIRT_VOL_NAME}\"}}" ||
+        error_exit "Failed to add the libvirt image id to peer-pods-cm configmap"
```

Comment:
I think UPDATE_PEERPODS_CM is ignored at creation time, shouldn't it avoid updating CM if set to false?

Reply:
I added, thanks for pointing it